### PR TITLE
Show commit status for releases

### LINKED ISSUE #29082: Actions Status: Release
### Feature Description

It would be great if we included the actions status for releases/tags for the underlying ref.

See related issue: https://github.com/go-gitea/gitea/issues/7773

### Screenshots

Mockup:
image

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #29082

grafik

### CODE REVIEW

File: templates/repo/release/list.tmpl [hunk 1]
```diff
@@ -5,90 +5,90 @@
 		{{template "base/alert" .}}
```
Comment (KN4CK3R) [inline templates/repo/release/list.tmpl:5]: Review this file without whitespace diff to spot the changes.

**Comment by yp05327** [comment]
I have a question. Is it necessary for a stable release.
I can understand add it for a pre-release or a draft release.
But for a stable release, when there's a red `x`, users may have a question:
is this release really stable? 😕

**Comment by delvh** [comment]
And I'd say rightfully so.

**Comment by KN4CK3R** [comment]
This PR now contains also a possible security fix. 

https://try.gitea.io/KN4CK3R/draft-test/releases hides draft versions
https://try.gitea.io/KN4CK3R/draft-test/releases/tag/v1.0 still shows the draft if you can guess the tag name
Don't know if this should get a "bug" or "security" label.

**Review by silverwind — APPROVED** [review]

**Comment by a1012112796** [comment]
a little wonder whether this change is necessary enough ? maybe user can add it in the release descraption manually or by ci.

**Comment by silverwind** [comment]
> maybe user can add it in the release descraption manually or by ci.

Why would one manually add a commit status? It does not make sense imho. If the release can be associated with a tag or commit, we should show the associated commit status, which can indicate problems for example when the release process is done by CI.

@KN4CK3R I think it's possible for release to not be associated to a commit, in such a case we should not render a status.

**Comment by KN4CK3R** [comment]
> @KN4CK3R I think it's possible for release to not be associated to a commit, in such a case we should not render a status.

Releases are tag bases and therefore must be associated to a commit.

**Comment by silverwind** [comment]
Okay, I guess it would still be good to add a check so that we do not break the template rendering if no commit can be determined.

**Comment by KN4CK3R** [comment]
Where should a check be placed? If there was no commit status found, nothing gets rendered.

**Comment by silverwind** [comment]
Before the subtemplate `repo/commit_statuses`, but if there is a check already, then it's fine. I did not check in detail.

**Comment by lunny** [comment]
Maybe it should be hidden if `Code` is disabled.

**Review by kdumontnu — APPROVED** [review]
Looks & works great in my tests :+1: 

If there is any concern about UI we could optionally move the status next to the commit hash so that it's clear it's related to the underlying ref. I'm happy either way.

**Pull request merged**